ExpensesWebApplication/Scripts/budget-strategy.js

```diff
@@ -0,0 +1,95 @@
+/** 
+ * @description Contains the business logic to handle the numeric calculations of budgets.
+ */
+class BudgetStrategy {
+
+    static #AVERAGE_DAYS_IN_A_MONTH_DECIMAL = 30.437;
+    static #MONTHS_IN_A_YEAR_DECIMAL = 12.0;
+    static #MONTHS_IN_A_HALFYEAR_DECIMAL = 6.0;
+    static #AVERAGE_WEEKS_PER_MONTH_DECIMAL = 4.345;
+    static #AVERAGE_BIWEEKS_PER_MONTH_DECIMAL = 4.345 / 2.0;
+
+    constructor() {
+
+    }
+
+
+    /**
+     * Returns the calculated cost of a single record over the course of "frequency" time.
+     * @param {decimal} cost The decimal money value in a single record at a specific frequency.
+     * @param {string} frequency How often the occurance of this cost occurs for the income or expense.
+     * @returns {decimal} The calculated cost of a single record over the course of "frequency" time.
+     */
+    GetMonthlySum(cost, frequency) {
+        // TODO - change this to enums and ints so the application can be multi-lingual.
```

Comment:
Potential enum declaration: 
public enum timePeriod {
Daily, Weekly, Monthly, Half-Annually, Annually
}

Reply:
That's a good solution for that.  Created ticket for work to be done.  I put your comment in the notes.